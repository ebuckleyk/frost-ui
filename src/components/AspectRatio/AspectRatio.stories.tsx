import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';

import { useImageOnLoad } from '../../hooks/useImageOnLoad';
import { AspectRatio } from './AspectRatio';

function AspectRatioDemo({ ratio }: { ratio: number | undefined }) {
  const { handleImageOnLoad, css } = useImageOnLoad();

  const style: { [key: string]: React.CSSProperties } = {
    wrap: {
      position: 'relative',
      width: 400,
      height: 400,
      margin: 'auto',
    },
    image: {
      position: 'absolute',
      top: 0,
      left: 0,
      width: `100%`,
      height: `100%`,
    },
  };

  return (
    <AspectRatio style={style} ratio={ratio}>
      <img
        style={{ ...style.image, ...(css.thumbnail as React.CSSProperties) }}
        src="https://wowslider.com/sliders/demo-31/data1/images/autumn1072827.jpg"
        alt="thumbnail"
      />
      <img
        onLoad={handleImageOnLoad}
        style={{ ...style.image, ...(css.fullSize as React.CSSProperties) }}
        src="https://wowslider.com/sliders/demo-31/data1/images/autumn1072827.jpg"
        alt="thumbnail"
      />
    </AspectRatio>
  );
}
type ComponentType = React.ComponentProps<typeof AspectRatio>;
const meta: Meta<ComponentType> = {
  component: AspectRatio,
  render: ({ ratio }) => <AspectRatioDemo ratio={ratio} />,
};

export default meta;

type Story = StoryObj<ComponentType>;
export const SixteenBy9: Story = {
  args: {
    ratio: 16 / 9,
  },
};
SixteenBy9.storyName = '16 / 9';

export const FourBy3: Story = {
  args: {
    ratio: 4 / 3,
  },
};

FourBy3.storyName = '4 / 3';
