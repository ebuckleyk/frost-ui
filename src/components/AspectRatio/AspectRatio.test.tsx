import * as React from 'react';
import { render } from '@testing-library/react';

import { AspectRatio } from './AspectRatio';

const style: { [key: string]: React.CSSProperties } = {
  wrap: {
    position: 'relative',
    width: 400,
    height: 400,
    margin: 'auto',
  },
  image: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: `100%`,
    height: `100%`,
  },
};
const Component = () => {
  return (
    <AspectRatio style={style} ratio={16}>
      <img
        style={{ ...style.image }}
        src="https://wowslider.com/sliders/demo-31/data1/images/autumn1072827.jpg"
        alt="thumbnail"
      />
    </AspectRatio>
  );
};

describe('AspectRatio', () => {
  it('should render AspectRatio component to match snapshot', () => {
    const result = render(<Component />);
    expect(result).toMatchSnapshot();
  });
});
