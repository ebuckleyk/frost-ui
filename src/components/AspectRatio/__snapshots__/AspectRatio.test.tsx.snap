// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AspectRatio > should render AspectRatio component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-radix-aspect-ratio-wrapper=""
        style="position: relative; width: 100%; padding-bottom: 6.25%;"
      >
        <div
          style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
        >
          <img
            alt="thumbnail"
            src="https://wowslider.com/sliders/demo-31/data1/images/autumn1072827.jpg"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-radix-aspect-ratio-wrapper=""
      style="position: relative; width: 100%; padding-bottom: 6.25%;"
    >
      <div
        style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
      >
        <img
          alt="thumbnail"
          src="https://wowslider.com/sliders/demo-31/data1/images/autumn1072827.jpg"
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
